File: exercism/ruby/bob/bob.rb
class Bob

  def self.hey(remark)
    return "Fine. Be that way!" if remark.match?(/^\s+$/) || remark.empty?
    return "Whoa, chill out!" if !remark.match?(/[a-z]/) && remark.match(/[A-Z]/)
    return "Sure." if remark.gsub(/\s/, "")[-1] == "?"
    return "Whatever."
  end
end

module BookKeeping
  VERSION = 1
end

File: exercism/javascript/bob/bob.js
function Bob() {}

Bob.prototype.hey = function(statement) {
  var cleaned = statement.replace(/\s/g, "");
  if (cleaned === cleaned.toUpperCase() && cleaned !== cleaned.toLowerCase()) {
    return "Whoa, chill out!";
  } else if (cleaned[cleaned.length - 1] === "?") {
    return "Sure.";
  } else if (cleaned === "") {
    return "Fine. Be that way!";
  } else {
    return "Whatever.";
  }
};

module.exports = Bob;
